source/dataman/CacheMan.h

```diff
@@ -41,7 +41,7 @@ class CacheItem
                                             const std::vector<size_t> &o)
     {
         std::vector<size_t> g;
-        for (int i = 0; i < p.size(); i++)
+        for (int i = 0; i < p.size(); ++i)
```

Comment:
Typically you want to think of the functors you pass to stl algorithms as small, compact, kernels.  It's not always the case but it's a good rule of thumb.  If the body of your loop is only a few lines then it's probably well suited to a functor in an stl algorithm.  If the iterations are independent of one another then it can also be beneficial to factor out to a functor applied by an stl algorithm.

However, if your loop body is large and complex with a lot of branching then it's probably best to leave in explicit loop form.